packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,336 @@
+import 'dart:convert';
+
+import 'package:flutter_tools/src/base/process.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:intl/intl.dart';
+import 'package:meta/meta.dart';
+import 'package:vm_snapshot_analysis/treemap.dart';
+
+import '../base/file_system.dart';
+import '../globals.dart' as globals;
+
+// This file can be in devtools_shared
+class SizeAnalyzer {
+  static const String kAotSizeJson = 'aot-size.json';
+
+  static String getAotSizeAnalysisJsonFile(String aotOutputPath) {
+    return globals.fs.path.join(aotOutputPath, kAotSizeJson);
+  }
+
+  static String getAotSizeAnalysisExtraGenSnapshotOption(String aotOutputPath) {
+    return '--print-instructions-sizes-to=${getAotSizeAnalysisJsonFile(aotOutputPath)}';
+  }
+
+  // Parse the output of unzip -v which shows the zip's contents' compressed sizes.
+  final RegExp _kGetUnzipOutput =
+      RegExp(r'^\s*\d+\s+[\w|:]+\s+(\d+)\s+.*  (.+)$');
+
+  Map<String, dynamic> analyzeApkSize(
+      {@required File apk, @required File aotSizeJson}) {
+    globals.printStatus('▒' * 80);
+    _printEntitySize(
+      '${apk.basename} (total compressed)',
+      apk.lengthSync() ~/ 1024,
+      0,
+    );
+    globals.printStatus('━' * 80);
+    final Directory tempApkContent = globals.fs.systemTempDirectory.createTempSync('flutter_tools.');
+    // TODO: implement Windows.
+    final String unzipOut = processUtils.runSync(<String>[
+      'unzip',
+      '-o',
+      '-v',
+      apk.path,
+      '-d',
+      tempApkContent.path
+    ]).stdout;
+    // We just want the the stdout printout. We don't need the files.
+    tempApkContent.deleteSync(recursive: true);
+
+    final Map<String, dynamic> apkAnalysisJson = _parseUnzipFile(
+      unzipOut,
+      json.decode(aotSizeJson.readAsStringSync()) as List<dynamic>,
+    );
+
+    final List<Map<String, dynamic>> firstLevelPaths = apkAnalysisJson['children'] as List<Map<String, dynamic>>;
```

Comment:
I'm surprised this as check works. I would have thought you would need to do call .toList() to make this a List<Map<String, dynamic>> instead of List<dynamic>